src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -307,6 +325,70 @@ private BoundExpression CreateMethodGroupConversion(CSharpSyntaxNode syntax, Bou
             return new BoundConversion(syntax, group, conversion, @checked: false, explicitCastInCode: isCast, constantValueOpt: ConstantValue.NotAvailable, type: destination, hasErrors: hasErrors) { WasCompilerGenerated = source.WasCompilerGenerated };
         }
 
+        private BoundExpression CreateTupleConversion(CSharpSyntaxNode syntax, BoundExpression source, Conversion conversion, TypeSymbol destination, DiagnosticBag diagnostics)
+        {
+            // We have a successful tuple conversion; rather than producing a node 
+            // which is a conversion on top of a tuple literal, tuple conversion is an element-wise conversion of arguments.
+
+            var sourceTuple = (BoundTupleLiteral)source;
+            var targetType = (NamedTypeSymbol)destination;
```

Comment:
This cast is unconditional.  Why not just make the parameter type `NamedTypeSymbol` to make the API more resilient to error?